services/datalad/datalad_service/common/git.py

```diff
@@ -26,6 +30,59 @@ def git_show(repo, committish, obj):
     return str(result)
 
 
+async def read_from_bytes_as_stream(data: bytes, chunk_size: int = 1024):
+    """
+    Reads from a bytes object as an async stream, yielding chunks.
+    """
+    buffer = io.BytesIO(data)
+    while True:
+        chunk = await asyncio.to_thread(buffer.read, chunk_size)
+        if not chunk:
+            break
+        yield chunk
```

Comment:
I might be missing something here, but I suspect you're going to spend more time with `asyncio.to_thread()` than you will with `data.__getitem__`:

```suggestion
    buffer = io.BytesIO(data)
    for i in range(0, len(data), chunk_size):
        yield data[i:i + chunk_size]
```

Edit: I guess this won't be necessary if you switch to blobio...